src/sentry/static/sentry/app/components/events/contexts/user.jsx

```diff
@@ -30,7 +39,16 @@ const UserContextType = React.createClass({
         <div className="pull-left">
           <Avatar user={user} size={96} gravatar={false} />
         </div>
-        <KeyValueList data={builtins} isContextData={false} />
+        <table className="key-value table">
```

Comment:
So, I see why you removed `KeyValueList` – because it wouldn't render your custom `Email` JSX above properly.

I feel a better solution is to modify `KeyValueList` so that if `value` is a React component, it should just spit out that component as-is (whereas primitives go through the existing path through `deviceNameMapper`).